tools/update-gke-infra.sh

```diff
@@ -0,0 +1,249 @@
+  echo "logs            Display logs of a Turbinia server or worker"
+  echo "show-config     Write the Turbinia configuration of an instance to STDOUT"
+  echo "status          Show the running status of server and workers"
+  echo "cordon          Cordon a cluster (Cordoning nodes is a Kubernetes mechanism to mark a node as “unschedulable”.)"
+  echo "uncordon        Uncordon a cluser (Cordoning nodes is a Kubernetes mechanism to mark a node as “unschedulable”.)"
+  echo "update-config   Update the Turbinia configuration of a Turbinia deployment from CONFIG_FILE, use -f"
+  echo "update-env      Update an environment variable on a container, use -k and -v"
+  echo "resize-cluster  Resize the number of nodes in the cluster."
+  echo
+}
+
+function check_gcloud {
+    if [ -z $GCLOUD ]
+    then
+        echo "gcloud not found, please install first"
+        exit 1
+    fi
+}
+
+function check_kubectl {
+    if [ -z $GCLOUD ]
+    then
+        echo "kubectl not found, please install first"
+        exit 1
+    fi
+}
+
+function show_infra {
+    $KUBECTL get pod -o=custom-columns=NAME:.metadata.name,STATUS:.status.phase,NODE:.spec.nodeName
+}
+
+function show_nodes {
+    $KUBECTL get nodes
+}
+
+function get_nodes {
+    NODES=$($KUBECTL get nodes --output=jsonpath={.items..metadata.name})
+}
+
+function cordon {
+    echo "Note this does not stop a cluster. Please resize the cluster to zero to prevent being billed."
+    # Show status
+    show_nodes
+
+    # Cordon all nodes
+    get_nodes
+
+    for NODE in $NODES
+    do
+        $KUBECTL cordon $NODE
+    done
+
+    # Show status
+    show_nodes
+}
+
+function uncordon {
+    # Show status
+    show_nodes
+
+    # Uncordon all nodes
+    get_nodes
+
+    for NODE in $NODES
+    do
+        $KUBECTL uncordon $NODE
+    done
+
+    # Show status
+    show_nodes
+}
+
+function show_container_logs {
+    show_infra
+    read -p 'Which container name? ' CONTAINER_NAME
+    $KUBECTL logs $CONTAINER_NAME
+}
+
+function show_config {
+    echo "Pulling Turbinia configuration from ConfigMap: turbinia-config"
+    $KUBECTL get configmap turbinia-config -o json | jq '.data.TURBINIA_CONF' | xargs | base64 -d
+}
+
+function update_config {
+    CONFIG_BASE64=`cat $CONFIG_FILE | base64 -w 0`
+    # Update ConfigMap with new Turbinia config
+    $KUBECTL create configmap turbinia-config --from-literal=TURBINIA_CONF=$CONFIG_BASE64 -o yaml --dry-run=client | $KUBECTL replace -f -
+}
+
+function show_deployment {
+    $KUBECTL get deployments
+}
+
+function update_env {
+    show_deployment
+
+    echo "Going to set environment variable $ENVKEY to $ENVVALUE"
+    read -p 'Which deployment? ' DEPLOYMENT
+
+    # Update the deployment
+    $KUBECTL set env deployment/$DEPLOYMENT $ENVKEY=$ENVVALUE
+
+}
+
+function resize_cluster {
+    echo "Resizing cluster $CLUSTER_NAME to $CLUSTER_SIZE nodes."
+    $GCLOUD container clusters resize $CLUSTER_NAME --num-nodes $CLUSTER_SIZE
```

Comment:
Does this respect the local lock files that we create, or will this resize regardless of that?

Reply:
It will resize regardless of it. Meant to be done by an admin when nothing is running but if it's risky, can remove that functionality though the script